src/otx/core/data/module.py

```diff
@@ -65,6 +67,16 @@ def __init__(
             self.config.test_subset.subset_name: self.config.test_subset,
         }
 
+        if self.config.auto_num_workers:
+            if self.config.device not in [DeviceType.gpu, DeviceType.auto]:
+                log.warning(
+                    "Only GPU device type support auto_num_workers. "
+                    f"Current deveice type is {self.config.device!s}. auto_num_workers is skipped.",
+                )
+            elif (num_workers := get_adaptive_num_workers()) is not None:
+                for subset_config in config_mapping.values():
+                    subset_config.num_workers = num_workers
```

Comment:
Add info message to user that `num_workers` changed by `auto_num_workers=True`.

Reply:
done at [3c84d46](https://github.com/openvinotoolkit/training_extensions/pull/2931/commits/3c84d46afb26d08a5d81469a9c502007e72ed61c)